File: BackEnd/backend/src/main/java/com/fitapp/backend/application/dto/routine/response/RoutineSummaryResponse.java
package com.fitapp.backend.application.dto.routine.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Set;

import com.fitapp.backend.infrastructure.persistence.entity.enums.DayOfWeek;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class RoutineSummaryResponse {
    private Long id;
    private String name;
    private Long userId;
    private String userName;
    private String description;
    private Long sportId;
    private String sportName;
    private Boolean isActive;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime lastUsedAt;
    private Set<DayOfWeek> trainingDays;
    private String goal;
    private Integer sessionsPerWeek;
    private Integer exerciseCount;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/routine/response/RoutineSummaryResponse.kt
package com.fitapp.appfit.response.routine.response

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class RoutineSummaryResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String?,
    @SerializedName("sportId") val sportId: Long?,
    @SerializedName("sportName") val sportName: String?,
    @SerializedName("isActive") val isActive: Boolean,
    @SerializedName("createdAt") val createdAt: String,
    @SerializedName("updatedAt") val updatedAt: String,
    @SerializedName("lastUsedAt") val lastUsedAt: String? = null,
    @SerializedName("trainingDays") val trainingDays: Set<String>,
    @SerializedName("goal") val goal: String,
    @SerializedName("sessionsPerWeek") val sessionsPerWeek: Int,
    @SerializedName("exerciseCount") val exerciseCount: Int
)